Makefile

```diff
@@ -66,12 +89,14 @@ metadata: lib all
 
 test: all
 	$(MAKE) -C haskell test PACKAGE=segmented-cfmm \
-		SEGMENTED_CFMM_PATH=../$(OUT)/segmented_cfmm_default.tz
+		SEGMENTED_CFMM_PATH=../$(OUT)/segmented_cfmm_default.tz \
+		STORAGE_PATH=../$(OUT)/storage_default.tz
```

Comment:
I'm confused about one thing: here you se these variables to build the storage, however in the `package.yaml` the `extra-source-file` lists only one of them and does so with a (different) hardcoded path:
```yaml
extra-source-files:
- test/segmented_cfmm.tz
```
AFAIK that file should be updated and this one should probably copy the output instead of setting these vars.

Reply:
The `extra-source-file` is updated as part of this PR: https://github.com/serokell/segmented-cfmm/commit/7b485366ffaecd5fdbeaff3fb1fab10bf86eaad5

I'm not sure where we get the path `test/segmented_cfmm.tz` from. Currently, if I build the project I get this warning: `Warning: File listed in segmented-cfmm.cabal file does not exist: test/segmented_cfmm.tz`.

I try to update the path to $(OUT) instead which is `../out/segmented_cfmm_default.tz`. That results in another warning: 

```Warning: 'extra-source-files: ../out/segmented_cfmm_default.tz' is a relative
path outside of the source tree. This will not work when generating a tarball
with 'sdist'.```

Not sure if that is ideal. So that's why I leave it the way it is. Let me know what you think about this.

Reply:
I see, it looks like we mixed two approaches here (env vars and extra-source-file) that don't work all that well together, I'll leave other comments to fix this.